File: 6-kyu-kata/steps-in-primes/python/steps_in_primes.py
# 6 kyu - Steps in Primes  [ ID: 5613d06cee1e7da6d5000055  (steps-in-primes) ]
# URL: https://www.codewars.com/kata/5613d06cee1e7da6d5000055
# Category: REFERENCE  |  Tags: MATHEMATICS | NUMBER THEORY
# ******************************************************************************
def step(g, m, n):
      i = m
      if m != 2 and g % 2:
          return None
      if m == 2 and is_prime(2+g):
          return [i, i+g]
      if not m % 2:
          i += 1
  
      while i <= n-g:
          if is_prime(i) and is_prime(i+g):
              return [i, i+g]
          i += 2
      return None
  
  
  def is_prime(i):
      for j in range(3, int(i**0.5+1), 2):
          if not i % j:
              return False
      return True

File: 6-kyu-kata/steps-in-primes/javascript/stepsInPrimes.js
// 6 kyu - Steps in Primes  [ ID: 5613d06cee1e7da6d5000055  (steps-in-primes) ]
// URL: https://www.codewars.com/kata/5613d06cee1e7da6d5000055
// Category: REFERENCE  |  Tags: MATHEMATICS | NUMBER THEORY
// *****************************************************************************
function isPrime(i) {
    for (j = 3; j * j <= i; j += 2) {
      if (!(i % j)) return false
    }
    return true
  }
  
  function step(g, m, n) {
    let i = m
    const primes = []
    if (m !== 2 && g % 2) return null
    if (m === 2) {
      primes.push(m)
      i = 3
    } else if (!(m % 2)) i += 1
  
    for (i; i <= n; i += 2) {
      if (!isPrime(i)) continue
      if (primes.includes(i - g)) return [i - g, i]
      primes.push(i)
    }
    return null
  }
// *****************************************************************************
// *****************************************************************************
function step(g, m, n) {
    if (g > 1 && g % 2) return null
    let primes = []
    let i = m
    if (m === 2) {
      primes.push(m)
      i = 3
    } else if (!(m % 2)) i += 1
  
    for (i; i <= n; i += 2) {
      let isPrime = true
      for (j = 3; j * j <= i; j += 2) {
        if (!(i % j)) {
          isPrime = false
          break
        }
      }
      if (isPrime && primes.includes(i - g)) return [i - g, i]
      isPrime && primes.push(i)
    }
    return null
  }

